pipelines/package_installer.go

```diff
@@ -0,0 +1,134 @@
+package pipelines
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"dagger.io/dagger"
+	"github.com/grafana/grafana-build/containers"
+	"github.com/grafana/grafana-build/executil"
+)
+
+type InstallerOpts struct {
+	PackageType     string
+	ConfigFiles     [][]string
+	AfterInstall    string
+	BeforeRemove    string
+	Depends         []string
+	EnvFolder       string
+	ExtraArgs       []string
+	AptDependencies []string
+}
+
+// Uses the grafana package given by the '--package' argument and creates a installer.
+// It accepts publish args, so you can place the file in a local or remote destination.
+func PackageInstaller(ctx context.Context, d *dagger.Client, args PipelineArgs, opts InstallerOpts) error {
+	packages, err := containers.GetPackages(ctx, d, args.PackageInputOpts)
+	if err != nil {
+		return err
+	}
+
+	debs := make(map[string]*dagger.File, len(packages))
+
+	for i, v := range args.PackageInputOpts.Packages {
+		var (
+			tarOpts = TarOptsFromFileName(v)
+			name    = filepath.Base(strings.TrimPrefix(strings.ReplaceAll(v, ".tar.gz", fmt.Sprintf(".%s", opts.PackageType)), "file://"))
+			fpmArgs = []string{
+				"fpm",
+				"--input-type=dir",
+				"--chdir=/pkg",
+				fmt.Sprintf("--output-type=%s", opts.PackageType),
+				"--vendor=\"Grafana Labs\"",
+				"--name=grafana",
+				"--description=Grafana",
```

Comment:
These two probably need to be grafana-enterprise if taropts.IsEnterprise is true